### PR TITLE
Implement `repeat` as a function that repeats strings, rename `repeat`->`repeat_row`

### PR DESCRIPTION
**Pull request by antifuchs**
This PR can be reviewed on a commit-by-commit basis:

1. First, we rename our experimental `repeat` function to `repeat_row`; this should be easy because it's undocumented and experimental.
2. Then, we implement `repeat` as the eponymous [postgresql string function](https://www.postgresql.org/docs/9.1/functions-string.html).

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/6309)

### CODE REVIEW

File: src/expr/src/scalar/func.rs [hunk 1]
```diff
@@ -4704,6 +4715,7 @@ pub enum VariadicFunc {
     RegexpMatch,
     HmacString,
     HmacBytes,
+    RepeatString,
```
Comment (benesch) [inline src/expr/src/scalar/func.rs:4718]: This seems like it wants to be a `BinaryFunc`, no?

File: src/expr/src/scalar/func.rs [hunk 2]
```diff
@@ -4066,6 +4066,17 @@ pub fn hmac_inner<'a>(
     Ok(Datum::Bytes(temp_storage.push_bytes(bytes)))
 }
 
+fn repeat_string<'a>(
+    datums: &[Datum<'a>],
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    Ok(Datum::String(temp_storage.push_string(
+        datums[0].unwrap_str().repeat(
+            usize::try_from(datums[1].unwrap_int32()).map_err(|_| EvalError::Int32OutOfRange)?,
```
Comment (benesch) [inline src/expr/src/scalar/func.rs:4075]: `repeat('a', -1)` actually seems to return `""` rather than an error in Postgres. Go figure. 🤷🏽

**Review by benesch — COMMENTED** [review]
Neat! @umanwizard any objection to the `repeat` -> `repeat_row` change? (Since you originally implement the function.)

File: src/expr/src/scalar/func.rs (hunk 2 above)
Comment (antifuchs) [reply src/expr/src/scalar/func.rs:4075]: Hahaha, oops! I'll make it so

File: src/expr/src/scalar/func.rs (hunk 1 above)
Comment (antifuchs) [reply src/expr/src/scalar/func.rs:4718]: True! Morning brain, will move.

**Comment by antifuchs** [comment]
Addressed @benesch's feedback - requesting that second review for the rename (:

**Review by benesch — APPROVED** [review]
LGTM! Can you also add a release note for the new function?

**Comment by umanwizard** [comment]
I have no objections to the rename. Furthermore, because `repeat` is marked as experimental, this does not violate our BC policy.

**Review by umanwizard — APPROVED** [review]

**Pull request merged**